addons/web_editor/models/ir_attachment.py

```diff
@@ -6,60 +6,73 @@
 from odoo import api, models, fields, tools
 
 SUPPORTED_IMAGE_MIMETYPES = ['image/gif', 'image/jpe', 'image/jpeg', 'image/jpg', 'image/png', 'image/svg+xml']
+SUPPORTED_VIDEO_MIMETYPES = ['video/mp4']
 SUPPORTED_IMAGE_EXTENSIONS = ['.gif', '.jpe', '.jpeg', '.jpg', '.png', '.svg']
+SUPPORTED_VIDEO_EXTENSIONS = ['.mp4']
 
 
 class IrAttachment(models.Model):
 
     _inherit = "ir.attachment"
 
     local_url = fields.Char("Attachment URL", compute='_compute_local_url')
-    image_src = fields.Char(compute='_compute_image_src')
-    image_width = fields.Integer(compute='_compute_image_size')
-    image_height = fields.Integer(compute='_compute_image_size')
+    src = fields.Char(compute='_compute_src')
+    filetype = fields.Char(compute='_compute_filetype', default='image')
+    width = fields.Integer(compute='_compute_size')
+    height = fields.Integer(compute='_compute_size')
+    thumbnail = fields.Binary("Thumbnail", attachment=False)
     original_id = fields.Many2one('ir.attachment', string="Original (unoptimized, unresized) attachment", index=True)
 
+    @api.depends('mimetype')
+    def _compute_filetype(self):
+        for attachment in self:
+            attachment.filetype = 'image' if attachment.mimetype in SUPPORTED_VIDEO_MIMETYPES else 'image'
```

Comment:
Yes, but it doesn't work yet so it's set to image for now. I need to change the upload directory to /video/... instead of /image/... and change the filetype when I manage to do that. Otherwise the source is not correct, and videos don't get rendered. WIP 😭